src/plugins/data-card/data-card-content.ts

```diff
@@ -239,3 +244,21 @@ export const DataCardContentModel = TileContentModel
   }));
 
 export interface DataCardContentModelType extends Instance<typeof DataCardContentModel> {}
+
+export function updateDataCardContentWithNewSharedModelIds(
+  content: DataCardContentModelType,
```

Comment:
Same concern here, this is typed to take an instance but it is being treated as a snapshot by code like `cloneDeep(content)`